### ISSUE TITLE
Problem: metsrw cannot process unicode characters

### ISSUE THREAD

**Issue by ross-spencer** [archivematica/Issues#295]
**Expected behaviour**

`metsrw` should be able to handle Unicode characters. 

**Current behaviour**

Python 2 only.

When we try to write Unicode characters we are getting the error: `ValueError: All strings must be XML compatible: Unicode or ASCII, no NULL bytes or control characters`

Related to: https://github.com/archivematica/Issues/issues/256 I believe there may be a more fundamental issue that needs to be fixed first, whereby `metsrw` cannot itself easily handle the Unicode characters we may want to supply it. The risk of not fixing this in `metsre` is that we may do unnecessary work in that #295. 

We can test this by changing around some of the fields in https://github.com/artefactual-labs/mets-reader-writer/blob/master/tests/constants.py, e.g. https://github.com/artefactual-labs/mets-reader-writer/blob/master/tests/constants.py#L8 - if this becomes `'Archivematica ā, ē, ī, ō, ū.'` then we experience the following Python 2 (only) failures (all is okay in Python 3):
```
tox
py27 installed: alabaster==0.7.11,atomicwrites==1.2.1,attrs==18.2.0,Babel==2.6.0,certifi==2018.8.24,chardet==3.0.4,coverage==4.5.1,docutils==0.14,funcsigs==1.0.2,future==0.16.0,idna==2.7,imagesize==1.1.0,Jinja2==2.10,lxml==4.2.5,MarkupSafe==1.0,metsrw==0.2.3,more-itertools==4.3.0,packaging==17.1,pathlib2==2.3.2,pluggy==0.7.1,py==1.6.0,Pygments==2.2.0,pyparsing==2.2.1,pytest==3.8.1,pytest-cov==2.6.0,pytz==2018.5,requests==2.19.1,scandir==1.9.0,six==1.11.0,snowballstemmer==1.2.1,Sphinx==1.8.1,sphinxcontrib-websupport==1.1.0,toml==0.9.6,tox==3.4.0,typing==3.6.6,urllib3==1.23,virtualenv==16.0.0
py27 run-test-pre: PYTHONHASHSEED='1476993255'
py27 runtests: commands[0] | pytest --cov-config .coveragerc --cov metsrw
========================================================================== test session starts ===========================================================================
platform linux2 -- Python 2.7.15rc1, pytest-3.8.1, py-1.6.0, pluggy-0.7.1
rootdir: /home/ross-spencer/git/artefactual-labs/mets-reader-writer, inifile:
plugins: cov-2.6.0
collected 100 items                                                                                                                                                      

tests/test_dependency_injection.py .F                                                                                                                              [  2%]
tests/test_fsentry.py ........................                                                                                                                     [ 26%]
tests/test_metadata.py ...................................                                                                                                         [ 61%]
tests/test_mets.py ..........................                                                                                                                      [ 87%]
tests/test_normative_structmap.py ..                                                                                                                               [ 89%]
tests/test_utils.py .                                                                                                                                              [ 90%]
tests/plugins/premisrw/test_premis.py ....FF....                                                                                                                   [100%]

================================================================================ FAILURES ================================================================================
___________________________________________________________ TestDependencyInjection.test_dependency_injection ____________________________________________________________

self = <tests.test_dependency_injection.TestDependencyInjection testMethod=test_dependency_injection>

    def test_dependency_injection(self):
        """Test the dependency injection (DI) infrastructure for metsrw plugins.
    
            - client: metsrw.FSEntry
            - services: classes for reading and writing metadata elements, e.g.,
              the PREMISObject class of metsrw.plugins.premisrw or other classes
              exposing the same interface.
            - injector: this test code or the code in metsrw/di.py which calls
              ``provide`` on the ``feature_broker`` singleton.
    
            The ``FSEntry`` class declares its dependency on the class attributes
            ``premis_object_class``, ``premis_event_class``, and
            ``premis_agent_class`` and further requires that these return classes
            with ``fromtree`` and ``serialize`` methods::
    
                >>> premis_object_class = Dependency(
                ...     has_methods('serialize'),
                ...     has_class_methods('fromtree'),
                ...     is_class)
    
            """
    
        # Clear the feature broker and then register/provide the premisrw
        # plugin classes (services) with the feature broker.
        feature_broker = metsrw.feature_broker
        assert len(feature_broker) == 3
        feature_broker.clear()
        assert not feature_broker
        feature_broker.provide('premis_object_class', premisrw.PREMISObject)
        feature_broker.provide('premis_event_class', premisrw.PREMISEvent)
        feature_broker.provide('premis_agent_class', premisrw.PREMISAgent)
        assert len(feature_broker) == 3
    
        # Create premisrw instances.
        compression_premis_event = premisrw.PREMISEvent(data=EX_COMPR_EVT)
        premis_events = [compression_premis_event]
        premis_agents = [premisrw.PREMISAgent(data=x)
                         for x in [EX_AGT_1, EX_AGT_2]]
        _, compression_program_version, archive_tool = (
            compression_premis_event.compression_details)
        premis_object = premisrw.PREMISObject(
            xsi_type=EX_PTR_XSI_TYPE,
            identifier_value=EX_PTR_IDENTIFIER_VALUE,
            message_digest_algorithm=EX_PTR_MESSAGE_DIGEST_ALGORITHM,
            message_digest=EX_PTR_MESSAGE_DIGEST,
            size=EX_PTR_SIZE,
            format_name=EX_PTR_FORMAT_NAME,
            format_registry_key=EX_PTR_FORMAT_REGISTRY_KEY,
            creating_application_name=archive_tool,
            creating_application_version=compression_program_version,
            date_created_by_application=EX_PTR_DATE_CREATED_BY_APPLICATION)
        transform_files = compression_premis_event.get_decompression_transform_files()
    
        # Create metsrw ``METSDocument`` and ``FSEntry`` instances.
        mets_doc = metsrw.METSDocument()
        fs_entry = metsrw.FSEntry(
            path=EX_PTR_PATH,
            file_uuid=EX_PTR_IDENTIFIER_VALUE,
            use=EX_PTR_PACKAGE_TYPE,
            type=EX_PTR_PACKAGE_TYPE,
            transform_files=transform_files,
            mets_div_type=EX_PTR_AIP_SUBTYPE)
        mets_doc.append_file(fs_entry)
    
        # Use the ``add_premis_...`` methods to add the PREMIS metadata
        # elements to the ``FSEntry`` instance. This will assert that each
        # PREMIS instance is of the correct type (e.g., that ``premis_object``
        # is an instance of ``FSEntry().premis_object_class``) and will call the
        # instance's ``serialize`` method and incorporate the resulting
        # ``lxml.etree._ElementTree`` instance into the ``FSEntry`` instance
        # appropriately.
        fs_entry.add_premis_object(premis_object)
        for premis_event in premis_events:
            fs_entry.add_premis_event(premis_event)
        for premis_agent in premis_agents:
>           fs_entry.add_premis_agent(premis_agent)

tests/test_dependency_injection.py:205: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
metsrw/fsentry.py:269: in add_premis_agent
    self.serialize_md_inst(md, self.premis_agent_class),
metsrw/fsentry.py:252: in serialize_md_inst
    return md_inst.serialize()
metsrw/plugins/premisrw/premis.py:133: in serialize
    return data_to_premis(self._data, self.premis_version)
metsrw/plugins/premisrw/premis.py:595: in data_to_premis
    return _data_to_lxml_el(data, 'premis', nsmap)
metsrw/plugins/premisrw/premis.py:490: in _data_to_lxml_el
    element, ns, nsmap, element_maker=element_maker, snake=snake))
metsrw/plugins/premisrw/premis.py:493: in _data_to_lxml_el
    ret = func(*args)
src/lxml/builder.py:224: in lxml.builder.ElementMaker.__call__
    ???
src/lxml/builder.py:173: in lxml.builder.ElementMaker.__init__.add_text
    ???
src/lxml/etree.pyx:1020: in lxml.etree._Element.text.__set__
    ???
src/lxml/apihelpers.pxi:711: in lxml.etree._setNodeText
    ???
src/lxml/apihelpers.pxi:699: in lxml.etree._createTextNode
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: All strings must be XML compatible: Unicode or ASCII, no NULL bytes or control characters

src/lxml/apihelpers.pxi:1439: ValueError
___________________________________________________________________ TestPREMIS.test_full_pointer_file ____________________________________________________________________

self = <tests.plugins.premisrw.test_premis.TestPREMIS testMethod=test_full_pointer_file>

    def test_full_pointer_file(self):
        """Test construction of a full METS pointer file. Assert that the
            resulting file is a valid METS file using the metsrw.validate function
            which validates both against the METS/PREMIS .xsd files and against an
            Archivematica-specific Schematron file.
            """
>       mw = self.create_test_pointer_file()

tests/plugins/premisrw/test_premis.py:88: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/plugins/premisrw/test_premis.py:78: in create_test_pointer_file
    mets_fs_entry.add_premis_agent(premisrw.data_to_premis(agent))
metsrw/plugins/premisrw/premis.py:595: in data_to_premis
    return _data_to_lxml_el(data, 'premis', nsmap)
metsrw/plugins/premisrw/premis.py:490: in _data_to_lxml_el
    element, ns, nsmap, element_maker=element_maker, snake=snake))
metsrw/plugins/premisrw/premis.py:493: in _data_to_lxml_el
    ret = func(*args)
src/lxml/builder.py:224: in lxml.builder.ElementMaker.__call__
    ???
src/lxml/builder.py:173: in lxml.builder.ElementMaker.__init__.add_text
    ???
src/lxml/etree.pyx:1020: in lxml.etree._Element.text.__set__
    ???
src/lxml/apihelpers.pxi:711: in lxml.etree._setNodeText
    ???
src/lxml/apihelpers.pxi:699: in lxml.etree._createTextNode
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: All strings must be XML compatible: Unicode or ASCII, no NULL bytes or control characters

src/lxml/apihelpers.pxi:1439: ValueError
___________________________________________________________________ TestPREMIS.test_pointer_file_read ____________________________________________________________________

self = <tests.plugins.premisrw.test_premis.TestPREMIS testMethod=test_pointer_file_read>

    def test_pointer_file_read(self):
        """Test that we can use the premisrw API to read values correctly from
            a METS/PREMIS pointer file.
            """
>       mw = self.create_test_pointer_file()

tests/plugins/premisrw/test_premis.py:100: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/plugins/premisrw/test_premis.py:78: in create_test_pointer_file
    mets_fs_entry.add_premis_agent(premisrw.data_to_premis(agent))
metsrw/plugins/premisrw/premis.py:595: in data_to_premis
    return _data_to_lxml_el(data, 'premis', nsmap)
metsrw/plugins/premisrw/premis.py:490: in _data_to_lxml_el
    element, ns, nsmap, element_maker=element_maker, snake=snake))
metsrw/plugins/premisrw/premis.py:493: in _data_to_lxml_el
    ret = func(*args)
src/lxml/builder.py:224: in lxml.builder.ElementMaker.__call__
    ???
src/lxml/builder.py:173: in lxml.builder.ElementMaker.__init__.add_text
    ???
src/lxml/etree.pyx:1020: in lxml.etree._Element.text.__set__
    ???
src/lxml/apihelpers.pxi:711: in lxml.etree._setNodeText
    ???
src/lxml/apihelpers.pxi:699: in lxml.etree._createTextNode
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: All strings must be XML compatible: Unicode or ASCII, no NULL bytes or control characters

src/lxml/apihelpers.pxi:1439: ValueError

-------- coverage: platform linux2, python 2.7.15-candidate-1 --------
Name                                  Stmts   Miss Branch BrPart  Cover
-----------------------------------------------------------------------
metsrw/__init__.py                       15      0      0      0   100%
metsrw/di.py                             79      5     28      3    91%
metsrw/exceptions.py                      4      0      0      0   100%
metsrw/fsentry.py                       206     13     77      4    93%
metsrw/metadata.py                      189      6     78      7    94%
metsrw/mets.py                          264     14    101     11    93%
metsrw/plugins/__init__.py                0      0      0      0   100%
metsrw/plugins/premisrw/__init__.py       8      0      0      0   100%
metsrw/plugins/premisrw/premis.py       368     21    156     12    94%
metsrw/plugins/premisrw/utils.py         37      0      4      0   100%
metsrw/utils.py                          17      0      4      0   100%
metsrw/validate.py                       67      7     16      5    86%
-----------------------------------------------------------------------
TOTAL                                  1254     66    464     42    93%

================================================================== 3 failed, 97 passed in 4.10 seconds ===================================================================
ERROR: InvocationError for command '/home/ross-spencer/git/artefactual-labs/mets-reader-writer/.tox/py27/bin/pytest --cov-config .coveragerc --cov metsrw' (exited with code 1)
```

`constants.py` contains values to help us test the PREMIS writing ability of the code. 

If we modify one of the mets tests as follows, we also get a similar error (diff first, then test output):

**NB** we modify: TestWholeMETS.test_full_mets to append `ū` to the following lines:
```
        file1 = metsrw.FSEntry('objects/object1ū.ext', file_uuid=str(uuid.uuid4()))
        file2 = metsrw.FSEntry('objects/object2ū.ext', file_uuid=str(uuid.uuid4()))
```

As follows:
```
ross-spencer@artefactual:~/git/artefactual-labs/mets-reader-writer$ git diff
diff --git a/tests/constants.py b/tests/constants.py
index 59cab28..b02b73b 100644
--- a/tests/constants.py
+++ b/tests/constants.py
@@ -2,9 +2,10 @@
 import uuid
 import metsrw.plugins.premisrw as premisrw
 
+# ā, ē, ī, ō, ū
 
 EX_AGT_1_IDENTIFIER_TYPE = 'preservation system'
-EX_AGT_1_IDENTIFIER_VALUE = 'Archivematica-1.6.1'
+EX_AGT_1_IDENTIFIER_VALUE = 'Archivematica'
 EX_AGT_1_NAME = 'Archivematica'
 EX_AGT_1_TYPE = 'software'
 EX_AGT_1 = (
diff --git a/tests/test_mets.py b/tests/test_mets.py
index 6acb119..cf9a715 100644
--- a/tests/test_mets.py
+++ b/tests/test_mets.py
@@ -363,8 +363,8 @@ class TestWholeMETS(TestCase):
 
     def test_full_mets(self):
         mw = metsrw.METSDocument()
-        file1 = metsrw.FSEntry('objects/object1.ext', file_uuid=str(uuid.uuid4()))
-        file2 = metsrw.FSEntry('objects/object2.ext', file_uuid=str(uuid.uuid4()))
+        file1 = metsrw.FSEntry('objects/object1ū.ext', file_uuid=str(uuid.uuid4()))
+        file2 = metsrw.FSEntry('objects/object2ū.ext', file_uuid=str(uuid.uuid4()))
         file1p = metsrw.FSEntry('objects/object1-preservation.ext', use='preservation', file_uuid=str(uuid.uuid4()), derived_from=file1)
         file2p = metsrw.FSEntry('objects/object2-preservation.ext', use='preservation', file_uuid=str(uuid.uuid4()), derived_from=file2)
         children = [file1, file2, file1p, file2p]
ross-spencer@artefactual:~/git/artefactual-labs/mets-reader-writer$ tox
py27 installed: alabaster==0.7.11,atomicwrites==1.2.1,attrs==18.2.0,Babel==2.6.0,certifi==2018.8.24,chardet==3.0.4,coverage==4.5.1,docutils==0.14,funcsigs==1.0.2,future==0.16.0,idna==2.7,imagesize==1.1.0,Jinja2==2.10,lxml==4.2.5,MarkupSafe==1.0,metsrw==0.2.3,more-itertools==4.3.0,packaging==17.1,pathlib2==2.3.2,pluggy==0.7.1,py==1.6.0,Pygments==2.2.0,pyparsing==2.2.1,pytest==3.8.1,pytest-cov==2.6.0,pytz==2018.5,requests==2.19.1,scandir==1.9.0,six==1.11.0,snowballstemmer==1.2.1,Sphinx==1.8.1,sphinxcontrib-websupport==1.1.0,toml==0.9.6,tox==3.4.0,typing==3.6.6,urllib3==1.23,virtualenv==16.0.0
py27 run-test-pre: PYTHONHASHSEED='712023176'
py27 runtests: commands[0] | pytest --cov-config .coveragerc --cov metsrw
========================================================================== test session starts ===========================================================================
platform linux2 -- Python 2.7.15rc1, pytest-3.8.1, py-1.6.0, pluggy-0.7.1
rootdir: /home/ross-spencer/git/artefactual-labs/mets-reader-writer, inifile:
plugins: cov-2.6.0
collected 100 items                                                                                                                                                      

tests/test_dependency_injection.py ..                                                                                                                              [  2%]
tests/test_fsentry.py ........................                                                                                                                     [ 26%]
tests/test_metadata.py ...................................                                                                                                         [ 61%]
tests/test_mets.py .................F........                                                                                                                      [ 87%]
tests/test_normative_structmap.py ..                                                                                                                               [ 89%]
tests/test_utils.py .                                                                                                                                              [ 90%]
tests/plugins/premisrw/test_premis.py ..........                                                                                                                   [100%]

================================================================================ FAILURES ================================================================================
______________________________________________________________________ TestWholeMETS.test_full_mets ______________________________________________________________________

self = <tests.test_mets.TestWholeMETS testMethod=test_full_mets>

    def test_full_mets(self):
        mw = metsrw.METSDocument()
        file1 = metsrw.FSEntry('objects/object1ū.ext', file_uuid=str(uuid.uuid4()))
        file2 = metsrw.FSEntry('objects/object2ū.ext', file_uuid=str(uuid.uuid4()))
        file1p = metsrw.FSEntry('objects/object1-preservation.ext', use='preservation', file_uuid=str(uuid.uuid4()), derived_from=file1)
        file2p = metsrw.FSEntry('objects/object2-preservation.ext', use='preservation', file_uuid=str(uuid.uuid4()), derived_from=file2)
        children = [file1, file2, file1p, file2p]
        objects = metsrw.FSEntry('objects', type='Directory', children=children)
        children = [
            metsrw.FSEntry('transfers', type='Directory', children=[]),
            metsrw.FSEntry('metadata/metadata.csv', use='metadata', file_uuid=str(uuid.uuid4())),
        ]
        metadata = metsrw.FSEntry('metadata', type='Directory', children=children)
        children = [
            metsrw.FSEntry('submissionDocumentation/METS.xml', use='submissionDocumentation', file_uuid=str(uuid.uuid4())),
        ]
        sub_doc = metsrw.FSEntry('submissionDocumentation', type='Directory', children=children)
        children = [objects, metadata, sub_doc]
        sip = metsrw.FSEntry('sipname-uuid', type='Directory', children=children)
        sip.add_dublin_core('<dublincore>sip metadata</dublincore>')
        file1.add_premis_object('<premis>object</premis>')
        file1.add_premis_event('<premis>event</premis>')
        file1.add_premis_agent('<premis>agent</premis>')
        rights = file1.add_premis_rights('<premis>rights</premis>')
        rights.replace_with(file1.add_premis_rights('<premis>newer rights</premis>'))
        dc = file1.add_dublin_core('<dublincore>metadata</dublincore>')
        dc.replace_with(file1.add_dublin_core('<dublincore>newer metadata</dublincore>'))
    
        mw.append_file(sip)
>       mw.write('full_metsrw.xml', fully_qualified=True, pretty_print=True)

tests/test_mets.py:393: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
metsrw/mets.py:308: in write
    root = self.serialize(fully_qualified=fully_qualified)
metsrw/mets.py:288: in serialize
    root.append(self._structmap())
metsrw/mets.py:228: in _structmap
    child = item.serialize_structmap(recurse=True)
metsrw/fsentry.py:406: in serialize_structmap
    recurse=recurse, normative=normative)
metsrw/fsentry.py:406: in serialize_structmap
    recurse=recurse, normative=normative)
metsrw/fsentry.py:396: in serialize_structmap
    el.attrib['LABEL'] = self.label
src/lxml/etree.pyx:2408: in lxml.etree._Attrib.__setitem__
    ???
src/lxml/apihelpers.pxi:570: in lxml.etree._setAttributeValue
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: All strings must be XML compatible: Unicode or ASCII, no NULL bytes or control characters

src/lxml/apihelpers.pxi:1439: ValueError

-------- coverage: platform linux2, python 2.7.15-candidate-1 --------
Name                                  Stmts   Miss Branch BrPart  Cover
-----------------------------------------------------------------------
metsrw/__init__.py                       15      0      0      0   100%
metsrw/di.py                             79      5     28      3    91%
metsrw/exceptions.py                      4      0      0      0   100%
metsrw/fsentry.py                       206     12     77      4    94%
metsrw/metadata.py                      189      6     78      7    94%
metsrw/mets.py                          264     14    101     11    93%
metsrw/plugins/__init__.py                0      0      0      0   100%
metsrw/plugins/premisrw/__init__.py       8      0      0      0   100%
metsrw/plugins/premisrw/premis.py       368     21    156     12    94%
metsrw/plugins/premisrw/utils.py         37      0      4      0   100%
metsrw/utils.py                          17      0      4      0   100%
metsrw/validate.py                       67      7     16      5    86%
-----------------------------------------------------------------------
TOTAL                                  1254     65    464     42    93%

================================================================== 1 failed, 99 passed in 6.33 seconds ===================================================================
ERROR: InvocationError for command '/home/ross-spencer/git/artefactual-labs/mets-reader-writer/.tox/py27/bin/pytest --cov-config .coveragerc --cov metsrw' (exited with code 1)
```

**Your environment (version of Archivematica, OS version, etc)**

`master` branch, testing on Ubuntu.

---
**For Artefactual use:**
Please make sure these steps are taken before moving this issue from Review to Verified in Waffle:

- All PRs related to this issue are properly linked 👍
- All PRs related to this issue have been merged 👍
- Test plan for this issue has been implemented and passed 👍
- Documentation regarding this issue has been written and it has been added to the release notes, if needed 👍

**Comment by sevein** [comment]
CC @cole ^^^

**Comment by sevein** [comment]
@ross-spencer here's a couple of thoughts.

    file1 = metsrw.FSEntry('objects/object1ū.ext', file_uuid=str(uuid.uuid4()))

This should be fixed in v0.3.0.

    E   ValueError: All strings must be XML compatible: Unicode or ASCII, no NULL bytes or control characters

My understanding here is that `constants.py` isn't using the `unicode_literals` future so you were defining a binary string. You should try `u'Archivematica ā, ē, ī, ō, ū.'`, I've tried and it prevents the error. Now the problem is that In Py2, premisrw is returning binary instead of unicode and some tests are going to fail, I think that's an oversight. I've submitted a new PR: https://github.com/artefactual-labs/mets-reader-writer/pull/57.
